### PR TITLE
more EBUS macro replacement

### PR DESCRIPTION
**Pull request by greerdv**
Signed-off-by: greerdv <greerdv@amazon.com>

## What does this PR do?
Replaces EBUS_EVENT style macros. The intention was to focus on code owned by sig-core, but seems to have included a little of sig-content and sig-network as well. The vast majority of changes were made using the regexes in #14099, but a few were done manually.

## How was this PR tested?
Built and ran tests.

### CODE REVIEW

**Review by cgalvan — APPROVED** [review]

**Review by moraaar — APPROVED** [review]

**Review by hultonha — APPROVED** [review]
A thing of beauty 😍 Thanks for updating @greerdv

File: Code/Framework/AzCore/AzCore/Slice/SliceSystemComponent.cpp [hunk 1]
```diff
@@ -57,10 +57,12 @@ namespace AZ
...
-        EBUS_EVENT(Data::AssetCatalogRequestBus, EnableCatalogForAsset, AZ::AzTypeInfo<AZ::SliceAsset>::Uuid());
-        EBUS_EVENT(Data::AssetCatalogRequestBus, AddExtension, "slice");
-        EBUS_EVENT(Data::AssetCatalogRequestBus, EnableCatalogForAsset, AZ::AzTypeInfo<AZ::DynamicSliceAsset>::Uuid());
-        EBUS_EVENT(Data::AssetCatalogRequestBus, AddExtension, "dynamicslice");
+        Data::AssetCatalogRequestBus::Broadcast(
+            &Data::AssetCatalogRequestBus::Events::EnableCatalogForAsset, AZ::AzTypeInfo<AZ::SliceAsset>::Uuid());
+        Data::AssetCatalogRequestBus::Broadcast(&Data::AssetCatalogRequestBus::Events::AddExtension, "slice");
+        Data::AssetCatalogRequestBus::Broadcast(
+            &Data::AssetCatalogRequestBus::Events::EnableCatalogForAsset, AZ::AzTypeInfo<AZ::DynamicSliceAsset>::Uuid());
+        Data::AssetCatalogRequestBus::Broadcast(&Data::AssetCatalogRequestBus::Events::AddExtension, "dynamicslice");
```
Comment (lumberyard-employee-dm) [inline Code/Framework/AzCore/AzCore/Slice/SliceSystemComponent.cpp:65]: Could you update this to a single lambda ... nevermind this is the legacy slice system.

**Review by lumberyard-employee-dm — APPROVED** [review]

File: Code/Framework/AzCore/AzCore/Slice/SliceSystemComponent.cpp (hunk 1 above)
Comment (greerdv) [reply Code/Framework/AzCore/AzCore/Slice/SliceSystemComponent.cpp:65]: I'd also prefer to make this change pure like for like replacement to minimize the chance of introducing an error.

File: Gems/LmbrCentral/Code/Source/Scripting/SpawnerComponent.cpp [hunk 2]
```diff
@@ -483,12 +483,14 @@ namespace LmbrCentral
...
+            SpawnerComponentNotificationBus::Event(
+                GetEntityId(), &SpawnerComponentNotificationBus::Events::OnEntitySpawned, ticket, currEntityId);
         }
 
-        EBUS_EVENT_ID(GetEntityId(), SpawnerComponentNotificationBus, OnSpawnEnd, ticket);
+        SpawnerComponentNotificationBus::Event(GetEntityId(), &SpawnerComponentNotificationBus::Events::OnSpawnEnd, ticket);
 
-        EBUS_EVENT_ID(GetEntityId(), SpawnerComponentNotificationBus, OnEntitiesSpawned, ticket, entityIds);
+        SpawnerComponentNotificationBus::Event(
+            GetEntityId(), &SpawnerComponentNotificationBus::Events::OnEntitiesSpawned, ticket, entityIds);
```
[the remark this answers is not in the document]
Comment (greerdv) [reply Gems/LmbrCentral/Code/Source/Scripting/SpawnerComponent.cpp:494]: I'd prefer to make this change pure like for like replacement to minimize the chance of introducing an error.

**Pull request merged**